src/pollers/mod.rs

```diff
@@ -100,8 +104,13 @@ pub trait ServerGatewayApis {
     ) -> Result<StartWorkflowExecutionResponse>;
 
     /// Fetch new work. Should block indefinitely if there is no work.
-    async fn poll_workflow_task(&self, task_queue: String)
-        -> Result<PollWorkflowTaskQueueResponse>;
+    async fn poll_task(&self, req: PollTaskRequest) -> Result<PollTaskResponse>;
+
+    /// Fetch new work. Should block indefinitely if there is no work.
```

Comment:
These docstrings should be more specific -- and I think we should get rid of the generic `poll_task`

Reply:
I agree, but let's do it as a follow up. As that would break lang side by doing that and it's better to coordinate that change with Roey.